src/openrct2/network/network.cpp

```diff
@@ -2134,7 +2134,11 @@ sint32 network_begin_client(const char *host, sint32 port)
 
 sint32 network_begin_server(sint32 port)
 {
-	return gNetwork.BeginServer(port);
+	char *address = nullptr;
+	if (strlen(gConfigNetwork.listen_address) > 0)
+		address = gConfigNetwork.listen_address;
+
+	return gNetwork.BeginServer(port, address);
```

Comment:
`network_begin_server` is just a C wrapper around `gNetwork.BeginServer`. So the parameter should be passed through along with port. Keep address and port together, pass `gConfigNetwork.listen_address` at the point `gConfigNetwork.port` is used. You may also want to add a command line argument for it too.